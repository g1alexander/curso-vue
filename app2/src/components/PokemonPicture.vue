<script setup lang="ts">
import { computed } from "vue";

const props = defineProps({
  pokemonId: {
    type: Number,
    required: true,
  },
  showPokemon: {
    type: Boolean,
    required: true,
    default: false,
  },
});

const imgSrc = computed(
  () =>
    `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/${props.pokemonId}.svg`
);
</script>

<template>
  <div class="pokemon-container">
    <img
      v-if="!showPokemon"
      :src="imgSrc"
      alt="pokemon"
      class="hidden-pokemon"
    />

    <img v-else :src="imgSrc" alt="pokemon" class="fade-in" />
  </div>
</template>
