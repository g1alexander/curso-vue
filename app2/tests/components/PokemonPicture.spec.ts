import { describe, expect, test } from "vitest";
import { shallowMount } from "@vue/test-utils";
import PokemonPicture from "../../src/components/PokemonPicture.vue";

describe("PokemonPicture component", () => {
  test("should to match snapchot", () => {
    const wrapper = shallowMount(PokemonPicture, {
      props: {
        pokemonId: 1,
        showPokemon: false,
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("debe mostrar la imagen oculta y el pokemon 100", () => {
    const wrapper = shallowMount(PokemonPicture, {
      props: {
        pokemonId: 100,
        showPokemon: false,
      },
    });

    const [img1, img2] = wrapper.findAll("img");
    const idPokemon = img1.attributes("src");

    expect(img1.exists()).toBe(true);

    expect(img2).toBe(undefined);

    expect(idPokemon.includes("100")).toBe(true);
  });

  test("debe de mostrar el pokemon si showPokemon:true", () => {
    const wrapper = shallowMount(PokemonPicture, {
      props: {
        pokemonId: 100,
        showPokemon: true,
      },
    });

    const img = wrapper.find("img");

    expect(img.classes("fade-in")).toBe(true);

    expect(img.classes("hidden-pokemon")).toBe(false);

    expect(img.exists()).toBe(true);
  });
});
