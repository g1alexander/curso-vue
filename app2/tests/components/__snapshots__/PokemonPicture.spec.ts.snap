// Vitest Snapshot v1

exports[`PokemonPicture component > should to match snapchot 1`] = `"<div class=\\"pokemon-container\\"><img src=\\"https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg\\" alt=\\"pokemon\\" class=\\"hidden-pokemon\\"></div>"`;
